### PR TITLE
Integrate gutenberg-mobile release 1.72.0

### PR DESCRIPTION
**Pull request by jhnstn**
## Description
This PR incorporates the 1.72.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR:

https://github.com/wordpress-mobile/gutenberg-mobile/pull/4668

Release Submission Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: build.gradle
Comment (hypest) [inline build.gradle]: Same as https://github.com/wordpress-mobile/WordPress-iOS/pull/18158#discussion_r829904208.

Comment (jhnstn) [reply build.gradle]: I replied over at the WordPress-iOS repo. `2095f19d1a7ed6d52b0f394f9579e66bb4876f0c` is ahead of the bundle steps and has the latest changes aside from the release note updates. But those will be included once we tag the release in Gutenberg Mobile and update `gutenbergMobileVersion` to point that tag.

Comment (hypest) [reply build.gradle]: Looks like we have a new bundle update at this point, let's point to the latest. Where I'm coming from: I prefer to start the merge train only after the whole PR chain is up-to-date and green. WDYT?

**Review by hypest — APPROVED** [review]
I will LGTM the PR @jhnstn as the changes make sense to me, and to enable you to move forward with the merge domino. The only thing to ask is to update the PRs so all pointers are the latest and wait for all-green in all PRs before the merge domino start. Thanks!

**Pull request merged**
